scripts/Install-AllSoftware.ps1

```diff
@@ -0,0 +1,47 @@
+# 1. Make sure the Microsoft App Installer is installed:
+#    https://www.microsoft.com/p/app-installer/9nblggh4nns1
+# 2. Run this script as administrator. If running scripts is blocked, you can temporarily unblock them by running 
+#    Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope Process
+
+function Check-IsElevated
+ {
+    $id = [System.Security.Principal.WindowsIdentity]::GetCurrent()
+    $p = New-Object System.Security.Principal.WindowsPrincipal($id)
+    if ($p.IsInRole([System.Security.Principal.WindowsBuiltInRole]::Administrator)) { 
+        Write-Output $true 
+    }      
+    else { 
+        Write-Output $false 
+    }   
+ }
+
+$policy = Get-ExecutionPolicy
+if ($policy -ne "Unrestricted"){
+    throw "Please set the execution policy to Unrestricted. You can temporarily unblock them by running
+    Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope Process"
+}
+elseif (-not(Check-IsElevated)){
+    throw "Please run this script as an administrator"
+ }
+else {
+    Write-Output "Installing apps"
+    $apps = @(
+    @{packageID = "icsharpcode.ILSpy" },
+    @{packageID = "Microsoft.DotNet.SDK.6" },
+    @{packageID = "Microsoft.VisualStudioCode" }
+    );
+
+    Foreach ($app in $apps) {
+    $listApp = winget list --exact -q $app.packageIDs
+    if (![String]::Join("", $listApp).Contains($app.packageID)) {
+        Write-Output "Installing: $($app.packageID)"
+        winget install -h $app.packageID -s winget 
+    }
+    else {
+        Write-Output "Skipping: $($app.packageID) (already installed)"
+    }
+    }
+
+    & "$PSScriptRoot\Get-VSPath.ps1" | Out-Null 
```

Comment:
What does this line do?

Reply:
It will check to see if the user as Visual Studio, along with its required workloads, installed, and if not, it will prompt the user and makes use of the Install-VS.ps1 script we already have.